src/js/models/RacingGame.js

```diff
@@ -48,17 +57,28 @@ class RacingGame {
 
       this.state.winner.push(item.name);
     });
-
-    return this.state.winner;
   }
 
-  play() {
+  runRound() {
+    this.state.movedInThisRound.length = 0;
     this.state.carList.forEach((car) => {
-      if (isAdvance()) {
+      if (this.isAdvance()) {
         car.go();
+        this.addCarInMovedList(car);
+        return;
       }
     });
```

Comment:
```suggestion
  runRound() {
    const roundResult = this.state.carList.map((car) => ({
      car,
      isProcessed: this.isAdvance(),
    }));

    roundResult.forEach(({ isProcessed, car }) => {
      if (isProcessed) {
        car.go();
      }
    });

    return roundResult
      .filter((result) => result.isProcessed)
      .map((result) => result.car.state.name);
  }
```

차가 전진했는지 여부를 별도의 상태로 관리하는게 나을까요??

`map`을 통해 `runRound`함수에서 전진한 차의 이름을 리턴해주는 방향은 어떨까요?

상태의 추가는 관리할 범위를 늘려서 지양해야한다고 생각해요! 🤔 
작성을 하다보니 `car` 모델에서 `state`를 통해서 차 내부의 속성들을 가져왔는데 메서드와 분리하기위해서 이렇게 사용하실까요?
어떤 의도로 분리하신건지 궁금합니다~

Reply:
차의 전진 여부를 상태로 관리하지 않는다는 생각은 못했어요!
함수에서 처리해주는 방법도 있었군요 👍👍
좋은 제안 감사합니다!


개인적으로는 state 안에 멤버 변수들을 저장하는게 깔끔해보이더라구요.
말씀하신대로 state가 붙으면 메서드가 아닌 변수구나! 하고 바로 알 수도 있구요!
MVC 패턴을 구현할때 view에 많은 양의 정보를 넘겨주는게 필요한 경우 state를 인자로 넘겨서 깔끔하게 처리해주려는 의도도 있었습니다.
다만 이렇게 하면 너무 뭉뚱그린 코드가 될 수도 있는게 살짝 걱정되긴 하네요. 🤔



```js
.map((result) => result.car.state.name);
```

car의 name의 경우 사전에 getter를 생성해놓아서 state에 직접적인 접근 없이 불러오도록 수정했습니다!

```js
.map((result) => result.car.name);
```